crates/astria-sequencer/src/genesis.rs

```diff
@@ -18,9 +24,54 @@ pub(crate) struct GenesisState {
 }
 
 impl GenesisState {
-    pub(crate) fn from_path<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
-        let file = File::open(path).context("failed to open file with genesis state")?;
-        serde_json::from_reader(&file).context("failed deserializing genesis state from file")
+    /// Load Account information from a CSV file
+    pub(crate) fn from_csv<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
+        let mut reader = Reader::from_path(path)?;
+        let mut records: Vec<Account> = Vec::new();
+
+        for result in reader.deserialize() {
+            let record: Account = result?;
+            records.push(record);
+        }
+
+        Ok(Self {
+            accounts: records,
+        })
+    }
+
+    /// Add the Account data from `GenesisState` to the `CometBFT` genesis.json file
+    pub(crate) fn propagate_accounts_to(&self, path: &str) -> anyhow::Result<()> {
+        // build the absolute path to the json file you want to add the accounts to
+        let mut home_path = env::var("HOME")?;
+        home_path.push_str(path);
+        let abs_destination_json_file_path = Path::new(&home_path);
+        let dest_file = File::open(abs_destination_json_file_path)
+            .context("failed to open destination genesis json file")?;
+
+        let mut dest_state: Value = serde_json::from_reader(&dest_file)
+            .context("failed deserializing genesis state from file")?;
+
+        // convert the accounts in GenesisState into a json Value
+        let mut json_map: serde_json::Map<String, Value> = serde_json::Map::new();
+        let mut accounts: Vec<Value> = Vec::new();
```

Comment:
If you implement (or derive) an appropriate `serde::ser::Serialize` trait for `GenesisState` all this can be avoided.